agent/agent.go

```diff
@@ -686,12 +686,14 @@ func (a *Agent) finishTask(ctx context.Context, tc *taskContext, status string,
 			setEndTaskCommand(tc, detail, "", "")
 		}
 		a.runEndTaskSync(ctx, tc, detail)
+		a.handleTimeoutAndOOM(ctx, tc, status)
```

Comment:
This has to run before `runPostTaskCommands` and before the `Task completed` log to preserve the original order of operations (same with below).

Reply:
moved